Userland/Libraries/LibAudio/Metadata.cpp

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2023, kleines Filmröllchen <filmroellchen@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "Metadata.h"
+#include <AK/Assertions.h>
+#include <LibCore/Version.h>
+
+namespace Audio {
+
+bool Person::is_artist() const
+{
+    return this->role == Person::Role::Artist
+        || this->role == Person::Role::Composer
+        || this->role == Person::Role::Conductor
+        || this->role == Person::Role::Lyricist
+        || this->role == Person::Role::Performer;
+}
+
+Optional<StringView> Person::name_for_role() const
+{
+    switch (this->role) {
+    case Role::Artist:
+        [[fallthrough]];
+    case Role::Performer:
+        return {};
+    case Role::Lyricist:
+        return "Lyricist"sv;
+    case Role::Conductor:
+        return "Conductor"sv;
+    case Role::Publisher:
+        return "Publisher"sv;
+    case Role::Engineer:
+        return "Engineer"sv;
+    case Role::Composer:
+        return "Composer"sv;
+    }
+    VERIFY_NOT_REACHED();
+}
+
+void Metadata::replace_encoder_with_serenity()
+{
+    auto version_or_error = Core::Version::read_long_version_string();
+    // Unset the encoder field in this case; we definitely want to replace the existing encoder field.
+    if (version_or_error.is_error())
+        this->encoder = {};
+    auto encoder_string = String::formatted("SerenityOS LibAudio {}", version_or_error.release_value());
+    if (encoder_string.is_error())
+        this->encoder = {};
+    this->encoder = encoder_string.release_value();
+}
+
+Optional<String> Metadata::first_artist() const
+{
+    return this->people.is_empty() ? Optional<String> {} : this->people.first().name;
+}
+
+ErrorOr<String> Metadata::all_artists(Optional<String> unknown_text_or_default, Optional<String> concatenate_with_or_default) const
+{
+    auto unknown_text = unknown_text_or_default.value_or(String::from_utf8_short_string("Unknown"sv));
+    auto concatenate_with = concatenate_with_or_default.value_or(String::from_utf8_short_string(", "sv));
+
+    // FIXME: This entire function could be similar to TRY(TRY(people.filter(...).try_map(...)).join(concatenate_with)).value_or(unknown_text) if these functional iterator transformers existed :^)
+    auto artists = this->people;
+    artists.remove_all_matching([](auto const& person) { return !person.is_artist(); });
+    if (artists.is_empty())
+        return unknown_text;
```

Comment:
This vector copy seems unnecessary. Why not just loop over `people` and slap a `if (!persion.is_artist()) continue;` at the top?